Source/SwiftLintCore/Models/Linter.swift

```diff
@@ -93,16 +102,26 @@ private extension Rule {
 
         let (disabledViolationsAndRegions, enabledViolationsAndRegions) = violations.map { violation in
             (violation, regions.first { $0.contains(violation.location) })
-        }.partitioned { _, region in
-            region?.isRuleEnabled(self) ?? true
+        }.partitioned { violation, region in
+            if let region {
+                return isEnabled(in: region, for: violation.ruleIdentifier)
+            }
+            return true
         }
 
+        let customRulesIDs: [String] = {
+             guard let customRules = self as? CustomRules else {
```

Comment:
Perhaps we could get rid of this by letting `Self.description.allIdentifiers` contain the custom IDs in case of custom rules.

Reply:
I couldn't see an easy way to customize `allIdentifiers`, as the RuleDescription currently can't see the configuration :-(

Reply:
Yeah, also what I struggled with. We'd need to move `allIdentifiers` into `Rule`, but this has other disadvantages. So let's not bother with that in this PR. I think it's good as it is. The special handling of custom rules is a broader topic we may address in follow-ups.